### PR TITLE
Speed up Admin Tab significantly

### PR DESCRIPTION
**Pull request by jrainville**
### What does the PR do

Iterates #16043
Is on top of #16279

Improves the speed of the Admin tab significantly by putting the different sections in Loaders, lazy loading them when needed.

It also fixes the members list speed a lot and also fixes the search.

It speeds up the permissions screen by using a ListView instead of a Repeater, that way, we don't instantiate all the items on start, only on scroll.

### Affected areas

Community admins panels and the Edit channel popup

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Architecture guidelines](../architecture-guide.md)

### Screenshot of functionality (including design for comparison)

:zap:  Video of how fast it is right now:

[fast-admin.webm](https://github.com/user-attachments/assets/27a57cd4-326a-4252-949a-78e411966997)

:snail: OLD version:

[slow-admin.webm](https://github.com/user-attachments/assets/6e97a6f5-4552-47ea-acf0-60b425a3a5a9)

As you can see, accessing the Admin panel is **way** faster now, since we lazy load. The only downside is that going to a particular tab, like the airdrop one, is a bit slower than before, because we load it when clicking on it, but a 0.5 second load is better than a 7+ seconds load on start.

You can also see, the members list search was excruciatingly slow. That is because it used the visibility of the elements instead of using the SortFilterProxyModel. It also didn't work well at all. It was case sensitive and only used the display name, so if you forgot that you put a nickname, it wouldn't work.

### Impact on end user

As seen in the video, the experience is way faster now and responsive.

### How to test

- Test all the admin panels and make sure they still navigate as expected (the content itself was not changed, just the loading, so in theory, no need to test the full airdrop for example)
- Test the edit channel popup to see if the permissions still show correctly

### Risk 

Tick **one**:
- [x] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

Worst case scenario is that one of the admin tabs is not accessible or the back buttons don't function. Or the permissions don't show correctly.

### CODE REVIEW

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (jrainville) [inline ui/app/AppLayouts/Communities/views/PermissionsView.qml:109]: This is the change that I'm less happy with.

I was able to convert this component to a ListView, and it worked well in the admin tab, however, when I checked the EditChannelPopup, it was all smooshed (no height). Also, it had a double scrollbar, which is not good.

I came up with this solution where if we are in the EditChannelPopup, we use the Repeater, and in the admin tab we use the ListView, but if you have an idea on how to make the ListView work in the popup, please let me know.

File: ui/imports/shared/stores/PermissionsStore.qml
Comment (jrainville) [inline ui/imports/shared/stores/PermissionsStore.qml:36]: Unused code

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:75]: Why is this here when empty?

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:58]: You could speed those proxy rules even further, by using `FastExpressionRole` with the `expectedRoles` property

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:93]: Make it `FastExpressionFilter`

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:103]: Same as above, `FastExpressionRole`s

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:131]: I think you could make this a binding instead of imperative code; ie. load the models when the `StackView`'s current page is xxx

File: ui/app/AppLayouts/Communities/panels/MembersTabPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/MembersTabPanel.qml:80]:
```suggestion
                filters: FastExpressionFilter {
```

File: ui/app/AppLayouts/Communities/panels/MembersTabPanel.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/MembersTabPanel.qml:83]: Hmm, why not use the `preferredDisplayName`? I thought it's part of the model now

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (caybro) [reply ui/app/AppLayouts/Communities/views/PermissionsView.qml:109]: Not good... you could use a ListView with expanded height (configurable), to avoid the double scrollbar. This double solution should be removed imho

**Review by caybro — COMMENTED** [review]
Nice speedups; some comments inline and I believe it could be sped up even more :)

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (jrainville) [reply ui/app/AppLayouts/Communities/views/PermissionsView.qml]: Ok, I was able to make it work!

Please review again. I also improved the speed even further by adding `prepareTokenModelForCommunityChat` that filters the permissions in Nim instead of doing it in the SortFilterProxyModel. The CreateChannelPopup opens way faster now

[edit-channel.webm](https://github.com/user-attachments/assets/1612caf5-82eb-4454-ac11-7a58517442cf)

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (caybro) [inline ui/app/AppLayouts/Communities/views/PermissionsView.qml:108]:
```qml
width: ListView.view.width
```

**Review by caybro — APPROVED** [review]
LGTM!

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (jrainville) [reply ui/app/AppLayouts/Communities/views/PermissionsView.qml:108]: I'm not sure why, when I use that, or `width: parent.width`, it ends up taking the whole width of the page, even though the ColumnLayout has `width: root.viewWidth`

image

**Comment by caybro** [comment]
The UI tests failure looks legit, needs fixing:
```
[2024-09-24T18:41:43.678Z] 1: >>> StoryBook pages with errors: ("/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-16363/storybook/pages/AirdropsSettingsPanelPage.qml")

[2024-09-24T18:41:44.045Z]     Test #1: PagesValidator ...................***Failed   23.84 sec
```

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (micieslak) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:121]: Wrapping here into `Loader` doesn't bring any benefit because it's already in `Component` which is already lazily instantiated on demand.

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (micieslak) [inline ui/app/AppLayouts/Communities/views/PermissionsView.qml:98]: as the list in layout, we shouldn't bind to `height` (https://doc.qt.io/qt-6/qml-qtquick-layouts-layout.html#details). `Layout.preferredHeight` can be instead.

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (micieslak) [inline ui/app/AppLayouts/Communities/views/PermissionsView.qml:110]: This `HoldingsSelectionModel` is sth with potentially big impact on performance (it's kind of proxy model but created when we had no c++ proxies). Probably should be updated and optimized at some point.

**Review by micieslak — CHANGES_REQUESTED** [review]
Great job! The speedup is really nice.

I have only one concern regarding architecture. Moving model transformations into ui components is basically against he guideline and should be treated as a last resort. The general rule is that UI component should be simple and take models ready to be displayed, with no need to do any transformations not directly related to rendering the view.

And in our case I think that `AirdropSettingsPanel` should stay as it was, taking ready to use data via
```qml
    required property var assetsModel
    required property var collectiblesModel
```
instead of taking
```qml
required property var communityTokens
```
which is processed then internally.

The key to achieve good performance is optimizing proxy roles and filters. There is a lot of room to do it as mentioned directly in comments.

Summarizing:
1. Use `RolesRenamingModel` to adjust role names
2. Use `FastExpressionRole` instead of `ExpressionRole`
3. Totally eliminate `FastExpressionRole` by introducing `FixedRole` proxy role (I can create it in a separate ticket, it will be helpful also in some other places)
4. If those steps above are not sufficient (but they already should) I have an idea for lazily initialized models for `AirdropsSettingsPanel` without changing the component.

```qml
assetsModel: LazyModel {
   SortFilterProxyModel {
        ...
   }
}
```
Such model would be lazily initialized when used first time (in this case when "New airdrop" tab is opened).

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (caybro) [reply ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:181]: Yeah I think it would be helpful; we use this pattern in many places

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (caybro) [reply ui/app/AppLayouts/Communities/views/PermissionsView.qml:110]: Yeah, I found this to be the slowest part when joining community / revealing the addresses

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (micieslak) [reply ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:181]: @jrainville, quick review and it's ready to use: https://github.com/status-im/status-desktop/pull/16405

Comment (jrainville) [reply ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml]: Ok, I'll wait for your PR to be merged to update

**Review by jrainville — COMMENTED** [review]
Thanks for the review @micieslak . The RenameRole indeed did make a huge difference already. I'll rebase on master and add the ConstantRole when yours is merged

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (jrainville) [reply ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml]: Done

**Review by micieslak — COMMENTED** [review]
Looks good in general, some smaller questions/remarks only.

File: ui/app/AppLayouts/Communities/views/CommunitySettingsView.qml
Comment (micieslak) [reply ui/app/AppLayouts/Communities/views/CommunitySettingsView.qml:182]: Ready to use: https://github.com/status-im/status-desktop/pull/16417 (ok, review needed first).

File: ui/app/AppLayouts/Communities/views/PermissionsView.qml
Comment (jrainville) [reply ui/app/AppLayouts/Communities/views/PermissionsView.qml:98]: It works! You must be some kind of QML genius :brain:

File: ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml
Comment (micieslak) [inline ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml:78]: It's not good place, please move it to `CommunitySettingsView`, into `sourceComponent: AirdropsSettingsPanel { ... }` :pray:

**Comment by micieslak** [comment]
Looks ready, one small detail regarding `Connections` and it seems ready to ship. The only question is if you want to wait and use https://github.com/status-im/status-desktop/pull/16417. @jrainville wdyt?

**Comment by jrainville** [comment]
@micieslak I tried using your component, but it doesn't work in the specific case of the `CommunitySettingsView` because we generate the buttons on the left for the different sections using properties on the Component itself. I managed to make it work by putting the properties in the Loaders, but with your Component, it doesn't work.

I don't doubt that we can find a way to make it work, I just think that since it's a bit more work, we might as well do it in another PR.

On another note, could one of you test the PR to see if the sections still work for you. Just doing some light tests before we can merge. Thanks

**Comment by micieslak** [comment]
> @micieslak I tried using your component, but it doesn't work in the specific case of the `CommunitySettingsView` because we generate the buttons on the left for the different sections using properties on the Component itself. I managed to make it work by putting the properties in the Loaders, but with your Component, it doesn't work.
> 
> I don't doubt that we can find a way to make it work, I just think that since it's a bit more work, we might as well do it in another PR.
> 
> On another note, could one of you test the PR to see if the sections still work for you. Just doing some light tests before we can merge. Thanks

Ahh, right, I see. The `goTo` mechanism is colliding here. It's some additional work to make it smoothly indeed. Le'ts use the current version, it's also good.

I tested in the app. Nothing suspicious found.

**Review by micieslak — APPROVED** [review]
Looks good! One comment left regarding some leftovers.

**Review by iurimatias — APPROVED** [review]

**Comment by jrainville** [comment]
Force merging as the tests passed, the report just failed. Michal did the basic tests

**Comment by anastasiyaig** [comment]
The display name in settings-messagin page is gone in this PR. i will fix that so this check will be present in tests running in PRs but we also need to fix the bug :D

**Pull request merged**
